src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +37,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
     (ActorFunc)BgSpot18Basket_Update,
     (ActorFunc)BgSpot18Basket_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7710.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7770.s")
+static ColliderJntSphItemInit sJntSphItemsInit[2] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+        { 0, { { 0, 2040, 0 }, 54 }, 100 },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { 1, { { 0, 1400, 0 }, 13 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x09, 0x20, COLSHAPE_JNTSPH },
+    2,
+    sJntSphItemsInit,
+};
+
+static s16 D_808B85C8[] = { 0x8000, 0x2AAA, 0xD555, 0x0000 };
+
+void func_808B7710(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+
+    Collider_InitJntSph(globalCtx, &this->colliderJntSph);
+    Collider_SetJntSph(globalCtx, &this->colliderJntSph, &this->dyna.actor, &sJntSphInit, &this->colliderJntSphItems);
+    this->dyna.actor.colChkInfo.mass = 0xFF;
+}
+
+void func_808B7770(BgSpot18Basket* this, GlobalContext* globalCtx, f32 arg2) {
+    static s16 D_808B85D0 = 0;
+    Vec3f acceleration;
+    Vec3f velocity;
+    Vec3f position;
+    f32 cosValue;
+    s32 i = 0;
+    f32 randomValue;
+    f32 sinValue;
+    s32 count = 2;
+
+    for (; i != count; i++) {
```

Comment:
Can this be:
```suggestion
    s32 i;
    f32 randomValue;
    f32 sinValue;
    s32 pad;

    for (i = 0; i < 2; i++) {
```
(and if so is the pad required)

I think it's more likely `i` and `count` (if real) were initialized in the for statement

Reply:
This actually doesn't match.  I tried several variations on for and while loops and ended up concluding that it required both values to be stored in temps that are initialized before the loop and it has to be !=.

Reply:
Can you do `for (i = 0, count = 2; i != count; i++) {` then?